src/components/molecules/NavBar/components/NavBarRadio/NavBarRadio.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useCallback, useMemo, useRef, useState } from "react";
+import { OverlayTrigger, Popover } from "react-bootstrap";
+import classNames from "classnames";
+
+import {
+  currentVenueSelectorData,
+  radioStationsSelector,
+} from "utils/selectors";
+import { hasElements } from "utils/types";
+
+import { useSelector } from "hooks/useSelector";
+import { useRadio } from "hooks/useRadio";
+
+import { RadioModal } from "components/organisms/RadioModal/RadioModal";
+
+import "./NavBarRadio.scss";
+
+export const NavBarRadio: React.FC = () => {
+  const venue = useSelector(currentVenueSelectorData);
```

Comment:
We should pass `venue` directly into this component as a prop rather than reading it from the state, something like this:

```suggestion
export interface NavBarRadioProps {
  venue: WithId<AnyVenue>,
}

export const NavBarRadio: React.FC<NavBarRadioProps> = ({ venue }) => {
```